connectors/webhook/src/main/java/io/camunda/connector/inbound/signature/strategy/URLAndBodyEncodingStrategy.java

```diff
@@ -6,39 +6,76 @@
  */
 package io.camunda.connector.inbound.signature.strategy;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.net.MediaType;
 import io.camunda.connector.api.inbound.webhook.WebhookProcessingPayload;
+import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
 import io.camunda.connector.inbound.utils.HttpWebhookUtil;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public final class URLAndBodyEncodingStrategy implements HMACEncodingStrategy {
-  @Override
-  public byte[] getBytesToSign(final WebhookProcessingPayload payload) throws IOException {
-    return (payload.requestURL() + extractSignatureData(payload)).getBytes();
+
+  private static String extractSignatureData(final WebhookProcessingPayload payload) {
+    return prepareForSignature(
+        payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers()));
   }
 
-  private static String extractSignatureData(final WebhookProcessingPayload payload)
-      throws IOException {
-    if (payload.rawBody() == null || payload.rawBody().length == 0) {
-      throw new NullPointerException(
-          "Can't extract signature data from body, because body is null");
+  private static String prepareForSignature(byte[] rawBody, String contentTypeHeader) {
+    if (MediaType.FORM_DATA.toString().equalsIgnoreCase(contentTypeHeader)) {
+      String bodyAsString =
+          URLDecoder.decode(new String(rawBody, StandardCharsets.UTF_8), StandardCharsets.UTF_8);
+      Map<String, String> strMap =
+          Arrays.stream(bodyAsString.split("&"))
+              .filter(Objects::nonNull)
+              .map(param -> param.split("="))
+              .collect(
+                  Collectors.toMap(param -> param[0], param -> param.length == 1 ? "" : param[1]));
+      return strMap.keySet().stream()
+          .sorted()
+          .map(key -> key.concat(strMap.get(key)))
+          .reduce(String::concat)
+          .orElse("");
     }
-    Map<String, String> signatureData =
-        HttpWebhookUtil.transformRawBodyToMap(
-            payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers()));
+    Object o = mapBytesToObject(rawBody);
+    if (o instanceof Map<?, ?> map) {
+      for (Map.Entry<?, ?> entry : map.entrySet()) {
+        if (entry.getKey() instanceof String) {
+          Map<String, String> strMap = (Map<String, String>) map;
+          return strMap.keySet().stream()
+              .sorted()
```

Comment:
do we know now why this is sorted ?

Reply:
Yes That's twilio specific, we have to keep it like this

Reply:
Then let's add a comment, because the next person will spend some time figuring this out :D

Reply:
@mathias-vandaele let's just add this comment and we're good to go :)